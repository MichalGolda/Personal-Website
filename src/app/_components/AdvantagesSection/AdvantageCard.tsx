import Image from "next/image";

export type AdvantageCardProps = {
  iconSrc: string;
  title: string;
  content: string;
};

const AdvantageCard: React.FC<AdvantageCardProps> = ({
  title,
  content,
  iconSrc,
}) => {
  return (
    <div className="flex flex-col gap-y-4 text-left justify-left">
      <Image
        src={iconSrc}
        alt={title}
        width={64}
        height={64}
        className="w-[48px] lg:w-[64px]"
      />
      <h4 className="font-bold text-lg md:text-xl lg:text-2xl text-secondary">
        {title}
      </h4>
      <p className="text-sm md:text-base text-body">{content}</p>
    </div>
  );
};

export default AdvantageCard;
