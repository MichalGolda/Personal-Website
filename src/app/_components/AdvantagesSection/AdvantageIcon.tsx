import Image from "next/image";

export type AdvantageIconProps = {
  src: string;
  alt: string;
};

const AdvantageIcon = ({ src, alt }: AdvantageIconProps) => (
  <Image
    src={src}
    alt={alt}
    width={64}
    height={64}
    className="w-[48px] lg:w-[64px]"
  />
);

export default AdvantageIcon;
