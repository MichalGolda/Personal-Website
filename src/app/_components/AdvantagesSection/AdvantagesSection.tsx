"use client";

import { useInViewSection } from "@/app/_context/inViewSectionContext";
import AdvantageCard from "./AdvantageCard";
import { Container, SectionHeading } from "@/app/_components";

export default function AdvantagesSection() {
  const { ref } = useInViewSection("#about-me");

  return (
    <section ref={ref} id="advantages" className="my-64">
      <Container>
        <SectionHeading headline="zalety" title="Dlaczego warto mi zaufać?" />
        <div className="my-16 gap-16 lg:grid-cols-3 flex flex-col gap-y-16 md:grid md:grid-cols-2 items-start">
          <AdvantageCard
            title="Przejrzyste rozliczenia"
            content="Przed rozpoczęciem projektu dokonuje wyceny w której zawarte są wszystkie wykonywane usługi. Dzięki, temu wiesz za co płacisz."
            iconSrc="/dollar.svg"
            data-aos="fade-up"
            data-aos-duration="1000"
          />
          <div className="col-span-2 order-2 lg:col-span-1 lg:order-[0]">
            <AdvantageCard
              title="Rozwiązania dopasowane do oczekiwań klienta"
              content="Każdy projekt rozpoczynam od zebrania szczegółowych informacji na temat tego jak ma wyglądać końcowe rozwiązanie. W trakcie realizacji konsultuje postępy po to żeby jak najlepiej wpasować się w twoje oczekiwania."
              iconSrc="/measure.svg"
              data-aos="fade-right"
              data-aos-duration="500"
            />
          </div>
          <AdvantageCard
            title="Realizacja w określonym terminie"
            content="Szczegółowo planuje etapy pracy dzięki czemu każde rozwiązanie jest gotowe na czas."
            iconSrc="/calendar.svg"
            data-aos="fade-left"
            data-aos-duration="1500"
          />
        </div>
      </Container>
    </section>
  );
}
