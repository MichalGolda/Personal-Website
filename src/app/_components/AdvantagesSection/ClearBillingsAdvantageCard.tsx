import AdvantageCard from "./AdvantageCard";
import AdvantageIcon from "./AdvantageIcon";

const DollarAdvantageIcon = () => (
  <AdvantageIcon src="/dollar.webp" alt="dollar" />
);

const ClearBillingsAdvantageCard = () => (
  <AdvantageCard
    title="Przejrzyste rozliczenia"
    content="Przed rozpoczęciem projektu dokonuje wyceny w której zawarte są wszystkie wykonywane usługi. Dzięki, temu wiesz za co płacisz."
    icon={<DollarAdvantageIcon />}
    data-aos="fade-right"
    data-aos-duration="1000"
  />
);

export default ClearBillingsAdvantageCard;
