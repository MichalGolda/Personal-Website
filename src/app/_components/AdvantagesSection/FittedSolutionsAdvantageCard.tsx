import AdvantageCard from "./AdvantageCard";
import AdvantageIcon from "./AdvantageIcon";

const MeasureAdvantageIcon = () => (
  <AdvantageIcon src="/measure.webp" alt="measure" />
);

const FittedSolutionsAdvantageCard = () => (
  <AdvantageCard
    title="Rozwiązania dopasowane do oczekiwań klienta"
    content="Każdy projekt rozpoczynam od zebrania szczegółowych informacji na temat tego jak ma wyglądać końcowe rozwiązanie. W trakcie realizacji konsultuje postępy po to żeby jak najlepiej wpasować się w twoje oczekiwania."
    icon={<MeasureAdvantageIcon />}
    data-aos="fade-right"
    data-aos-duration="1500"
  />
);

export default FittedSolutionsAdvantageCard;
