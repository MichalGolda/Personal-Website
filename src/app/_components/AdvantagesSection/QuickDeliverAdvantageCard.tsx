import AdvantageCard from "./AdvantageCard";
import AdvantageIcon from "./AdvantageIcon";

const CalendarAdvantageIcon = () => (
  <AdvantageIcon src="/calendar.webp" alt="calendar" />
);

const QuickDeliverAdvantageCard = () => (
  <AdvantageCard
    title="Realizacja w określonym terminie"
    content="Szczegółowo planuje etapy pracy dzięki czemu każde rozwiązanie jest gotowe na czas."
    icon={<CalendarAdvantageIcon />}
    data-aos="fade-right"
    data-aos-duration="2000"
  />
);

export default QuickDeliverAdvantageCard;
