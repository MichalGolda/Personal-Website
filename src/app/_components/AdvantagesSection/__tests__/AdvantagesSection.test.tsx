import "@/app/_mocks/intersectionObserver.mock"
import "@testing-library/jest-dom"
import { render, screen } from "@testing-library/react"
import AdvantagesSection from "../AdvantagesSection"

describe("AdvantagesSection component", () => {
    it("renders without errors", () => {
        render(<AdvantagesSection data-testid="advantages-section" />)

        const section = screen.getByTestId("advantages-section")
        const title = screen.getByText("Dlaczego warto mi zaufaÄ‡?")
        const headline = screen.getByText("zalety")

        expect(section).toBeInTheDocument()
        expect(section.tagName).toBe("SECTION")
        expect(section).toHaveAttribute("id", "advantages")

        expect(title).toBeInTheDocument()
        expect(title.tagName).toBe("H2")

        expect(headline).toBeInTheDocument()
        expect(headline.tagName).toBe("SPAN")
    })

    it("renders unchanged", () => {
        const { container } = render(<AdvantagesSection />)
        
        expect(container).toMatchSnapshot()
    })
})