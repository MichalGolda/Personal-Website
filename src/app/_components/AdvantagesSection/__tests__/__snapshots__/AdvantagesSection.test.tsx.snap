// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdvantagesSection component renders unchanged 1`] = `
<div>
  <section
    class="my-64"
    id="advantages"
  >
    <div
      class="md:max-w-[1024px] lg:max-w-[1320px] w-full mx-auto md:px-16 px-4 "
    >
      <div
        class="text-center max-w-3xl mx-auto"
      >
        <span
          class="text-primary font-semibold text-md md:text-lg lg:text-xl uppercase"
        >
          zalety
        </span>
        <h2
          class="text-2xl md:text-3xl lg:text-4xl text-secondary font-bold"
        >
          Dlaczego warto mi zaufać?
        </h2>
      </div>
      <div
        class="my-16 gap-16 lg:grid-cols-3 flex flex-col gap-y-16 md:grid md:grid-cols-2 items-start"
      >
        <div
          class="flex flex-col gap-y-4 text-left justify-left"
          data-aos="fade-right"
          data-aos-duration="1000"
        >
          <img
            alt="Przejrzyste rozliczenia"
            class="w-[48px] lg:w-[64px]"
            data-nimg="1"
            decoding="async"
            height="64"
            loading="lazy"
            src="/dollar.svg"
            style="color: transparent;"
            width="64"
          />
          <h4
            class="font-bold text-lg md:text-xl lg:text-2xl text-secondary"
          >
            Przejrzyste rozliczenia
          </h4>
          <p
            class="text-sm md:text-base text-body"
          >
            Przed rozpoczęciem projektu dokonuje wyceny w której zawarte są wszystkie wykonywane usługi. Dzięki, temu wiesz za co płacisz.
          </p>
        </div>
        <div
          class="col-span-2 order-2 lg:col-span-1 lg:order-[0]"
        >
          <div
            class="flex flex-col gap-y-4 text-left justify-left"
            data-aos="fade-right"
            data-aos-duration="1500"
          >
            <img
              alt="Rozwiązania dopasowane do oczekiwań klienta"
              class="w-[48px] lg:w-[64px]"
              data-nimg="1"
              decoding="async"
              height="64"
              loading="lazy"
              src="/measure.svg"
              style="color: transparent;"
              width="64"
            />
            <h4
              class="font-bold text-lg md:text-xl lg:text-2xl text-secondary"
            >
              Rozwiązania dopasowane do oczekiwań klienta
            </h4>
            <p
              class="text-sm md:text-base text-body"
            >
              Każdy projekt rozpoczynam od zebrania szczegółowych informacji na temat tego jak ma wyglądać końcowe rozwiązanie. W trakcie realizacji konsultuje postępy po to żeby jak najlepiej wpasować się w twoje oczekiwania.
            </p>
          </div>
        </div>
        <div
          class="flex flex-col gap-y-4 text-left justify-left"
          data-aos="fade-right"
          data-aos-duration="2000"
        >
          <img
            alt="Realizacja w określonym terminie"
            class="w-[48px] lg:w-[64px]"
            data-nimg="1"
            decoding="async"
            height="64"
            loading="lazy"
            src="/calendar.svg"
            style="color: transparent;"
            width="64"
          />
          <h4
            class="font-bold text-lg md:text-xl lg:text-2xl text-secondary"
          >
            Realizacja w określonym terminie
          </h4>
          <p
            class="text-sm md:text-base text-body"
          >
            Szczegółowo planuje etapy pracy dzięki czemu każde rozwiązanie jest gotowe na czas.
          </p>
        </div>
      </div>
    </div>
  </section>
</div>
`;
